stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -460,6 +461,37 @@ TypeRefBuilder::getBuiltinTypeInfo(const TypeRef *TR) {
   return nullptr;
 }
 
+class BuiltinTypeDescriptorImpl : public BuiltinTypeDescriptorInterface {
+  RemoteRef<BuiltinTypeDescriptor> BTD;
+  TypeRefBuilder &Builder;
+
+public:
+  BuiltinTypeDescriptorImpl(RemoteRef<BuiltinTypeDescriptor> BTD,
+                            TypeRefBuilder &Builder)
+      : BuiltinTypeDescriptorInterface(BTD->Size, BTD->getAlignment(),
+                                       BTD->Stride, BTD->NumExtraInhabitants,
+                                       BTD->isBitwiseTakable()),
+        BTD(BTD), Builder(Builder) {}
+
+  ~BuiltinTypeDescriptorImpl() override {}
+
+  StringRef readMangledTypeName() override {
+    return Builder.getTypeRefString(Builder.readTypeRef(BTD, BTD->TypeName));
+  };
+};
+
+std::unique_ptr<BuiltinTypeDescriptorInterface>
+TypeRefBuilder::getBuiltinTypeInfoInterface(const TypeRef *TR) {
+  if (DF) 
+    if (auto descriptor = DF->getBuiltinTypeInfoInterface(TR))
+      return descriptor;
+
+  auto BTI = getBuiltinTypeInfo(TR);
```

Comment:
Why isn't the second part of this function encapsulated in a second TypeDescriptorFinder that finds types in ReflectionMetadata?

Reply:
Ok, I separated this into its own class. I have it on its own NFC commit so it might be easier to review it that way